x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/MlNativeDataFrameAnalyticsIntegTestCase.java

```diff
@@ -173,12 +176,20 @@ protected void assertProgress(String id, int reindexing, int loadingData, int an
         assertThat(progress.get(3).getProgressPercent(), equalTo(writingResults));
     }
 
-    protected SearchResponse searchStoredProgress(String id) {
+    protected SearchResponse searchStoredProgress(String jobId) {
+        String docId = DataFrameAnalyticsTask.progressDocId(jobId);
         return client().prepareSearch(AnomalyDetectorsIndex.jobStateIndexPattern())
-            .setQuery(QueryBuilders.idsQuery().addIds(DataFrameAnalyticsTask.progressDocId(id)))
+            .setQuery(QueryBuilders.idsQuery().addIds(docId))
             .get();
     }
 
+    protected void assertInferenceModelPersisted(String jobId) {
+        SearchResponse searchResponse = client().prepareSearch(InferenceIndexConstants.LATEST_INDEX_NAME)
+            .setQuery(QueryBuilders.prefixQuery(TrainedModelConfig.MODEL_ID.getPreferredName(), jobId + "-"))
```

Comment:
You can use a `term` query against the `tags` field.
Something like the following.
```
QueryBuilders.boolQuery().filter(QueryBuilders.termQuery(TrainedModelConfig.TAGS.getPreferredName(), jobId))
```